src/Handler/NewsletterSubscriptionHandler.php

```diff
@@ -63,85 +53,134 @@ public function __construct(
         $this->listId = $listId;
     }
 
-    /**
-     * @param string $email
-     */
-    public function subscribe($email)
+    public function subscribe(string $email): void
     {
         $customer = $this->customerRepository->findOneBy(['email' => $email]);
 
-        if ($customer instanceof CustomerInterface) {
-            $this->updateCustomer($customer);
-        } else {
-            $this->createNewCustomer($email);
+        if (!$customer instanceof CustomerInterface) {
+            $customer = $this->createNewCustomer($email);
         }
 
-        $response = $this->mailChimp->get('lists/' . $this->listId . '/members/' . $this->getEmailHash($email));
-        Assert::keyExists($response, 'status');
+        $this->addMailchimpData($email);
 
-        if ($response['status'] === Response::HTTP_NOT_FOUND) {
-            $this->exportNewEmail($email);
+        $customer->setSubscribedToNewsletter(true);
+        $this->customerManager->flush();
+    }
+
+    public function getValidMailchimpListIds(): array
+    {
+        $responseArray = $this->mailChimp->get('/lists');
```

Comment:
Maybe move this hardcoded strings like "lists" to consts? Looks like we are using it more than once.